internal/remotestate/remote_state.go

```diff
@@ -127,3 +143,39 @@ func (remote *RemoteState) GenerateOpenTofuCode(opts *options.TerragruntOptions)
 
 	return remote.Config.GenerateOpenTofuCode(opts, backendConfig)
 }
+
+func (remote *RemoteState) pullState(ctx context.Context, opts *options.TerragruntOptions) (string, error) {
+	opts.Logger.Debugf("Pulling state from %s backend", remote.BackendName)
+
+	args := []string{tf.CommandNameState, tf.CommandNamePull}
+
+	output, err := tf.RunCommandWithOutput(ctx, opts, args...)
+	if err != nil {
+		return "", err
+	}
+
+	opts.Logger.Debugf("Creating temporary state file for migration")
+
+	file, err := os.CreateTemp("", "*.tfstate")
+	if err != nil {
+		return "", errors.New(err)
+	}
+
+	defer func() {
+		file.Close() // nolint: errcheck
+	}()
+
+	if _, err := file.Write(output.Stdout.Bytes()); err != nil {
+		return "", errors.New(err)
```

Comment:
NIT: Avoid doing this. Just return the following here:

```go
return file.Name(), errors.New(err)
```

The caller can decide if it's still worth it to read the state file, even if an error occurred.

Reply:
Thanks, good point.